### PR TITLE
REF: implement _unbox to de-duplicate unwrapping

### PR DESCRIPTION
**Pull request by jbrockmendel**
Ultimately we want these _validate_foo methods to be whittled down to only one or two methods.

### CODE REVIEW

File: pandas/core/arrays/datetimelike.py [hunk 1]
```diff
@@ -52,6 +52,8 @@
 from pandas.tseries import frequencies
 from pandas.tseries.offsets import DateOffset, Tick
 
+DTScalar = Union[Period, Timestamp, Timedelta, NaTType]
```
Comment (jreback) [inline pandas/core/arrays/datetimelike.py:55]: would put in _typing and call it DatetimeScalar

Comment (jbrockmendel) [reply pandas/core/arrays/datetimelike.py:55]: ok

Comment (jbrockmendel) [reply pandas/core/arrays/datetimelike.py:55]: Actually looks like we already have DatetimeLikeScalar which just needs to have NaTType added to it

Comment (jbrockmendel) [reply pandas/core/arrays/datetimelike.py]: updated+green

File: pandas/core/arrays/datetimelike.py [hunk 2]
```diff
@@ -52,6 +52,8 @@
 from pandas.tseries import frequencies
 from pandas.tseries.offsets import DateOffset, Tick
 
+DTScalarOrNaT = Union[DatetimeLikeScalar, NaTType]
```
Comment (jreback) [inline pandas/core/arrays/datetimelike.py:55]: can't we add NaTType to the DatetimeLikeScalar?

Comment (jbrockmendel) [reply pandas/core/arrays/datetimelike.py:55]: That would weaken the typing on the places where we currently use DatetimeLikeScalar

**Pull request merged**